### ISSUE TITLE
💅 the len is 32 but the index is 1208

### ISSUE THREAD

**Issue by roderik** [biomejs/biome#7368]
### Environment information

```bash
$ biome lint --diagnostic-level=error --max-diagnostics=5000 --write .
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:439:33
Thread Name: biome::workspace_worker_10
Message: index out of bounds: the len is 20 but the index is 1208

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:439:33
Thread Name: biome::workspace_worker_15
Message: index out of bounds: the len is 48 but the index is 1208

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:439:33
Thread Name: biome::workspace_worker_6
Message: index out of bounds: the len is 32 but the index is 1208

apps/app/components/layouts/box/lib/BoxLayout.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 48 but the index is 1208

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.


apps/app/components/layouts/centered/lib/CenteredLayout.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 20 but the index is 1208

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.


apps/app/components/layouts/split/lib/SplitLayout.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 32 but the index is 1208

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.


Checked 2919 files in 2s. No fixes applied.
```

### Rule name

unknown

### Playground link

not possible (biome v1 worked, upgraded to v2) + ultracite upgrade

### Expected result

Work

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by scrgl** [comment]
Hi, I have the same problem in my code. I took one of my files where biome was failing and gradually removed parts of the code until I found the exact culprit. Here’s the minimal repro:

```
import type { HTMLAttributes, ReactNode } from 'react';

// ❌ biome fails with interface
// interface Props extends HTMLAttributes<HTMLElement> {
//   endSlot?: ReactNode;
// }

// ✅ biome works with type
type Props = HTMLAttributes<HTMLElement> & {
  endSlot?: ReactNode;
};

export const Test = ({ endSlot }: Props) => {
  return <div>{endSlot && <div>{endSlot}</div>}</div>;
};
```


It seems that when interface is used to extend HTMLAttributes, biome fails, but switching to a type with an intersection works correctly.

That said, I’m still seeing issues in some other files even after replacing interface with type, so there are probably additional problems in biome beyond this particular case.

**Comment by siketyan** [comment]
@scrgl That might be a duplicate of #7324
